From 554046a5ccdf81e66d6cf3a8be2a080bf7505ffe Mon Sep 17 00:00:00 2001
From: Jim Stichnoth <jstichnoth@mythtv.org>
Date: Sun, 19 Apr 2015 12:10:29 -0700
Subject: [PATCH] Subtitles: Don't reset subtitle state after seeking.

After a seek, ac3a988 caused the subtitle-enabled state to be reset to
the desired state at playback start, rather than left unchanged.

(cherry picked from commit 26f6437b874d84c87a52df6d33e449c0d5d9a4bc)
---
 mythtv/libs/libmythtv/mythplayer.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/mythplayer.cpp b/mythtv/libs/libmythtv/mythplayer.cpp
index 2995169..a62ab31 100644
--- a/mythtv/libs/libmythtv/mythplayer.cpp
+++ b/mythtv/libs/libmythtv/mythplayer.cpp
@@ -1377,6 +1377,7 @@ void MythPlayer::DisableCaptions(uint mode, bool osd_msg)
 
     QMutexLocker locker(&osdLock);
 
+    textDesired = false;
     QString msg = "";
     if (kDisplayNUVTeletextCaptions & mode)
         msg += tr("TXT CAP");
@@ -1413,7 +1414,8 @@ void MythPlayer::DisableCaptions(uint mode, bool osd_msg)
 void MythPlayer::EnableCaptions(uint mode, bool osd_msg)
 {
     QMutexLocker locker(&osdLock);
-    QString msg;
+    textDesired = true;
+    QString msg = "";
     if ((kDisplayCC608 & mode) || (kDisplayCC708 & mode) ||
         (kDisplayAVSubtitle & mode) || kDisplayRawTextSubtitle & mode)
     {
